### PR TITLE
feature: Publish orbs with codacy versioning and cleanup build config

### PR DESCRIPTION
**Pull request by h314to**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,120 +1,116 @@
...
-# add your orb below, to be used in integration tests (note: a @dev:alpha
-# release must exist; you'll need to publish manually at least once)
-# you can use scripts/publish-alpha.sh to publish a @dev:alpha release
-orbs:
-  codacy: codacy/base@dev:alpha
-  orb-tools: circleci/orb-tools@8.5.0
+references:
+  git_job: &git_job
+    docker:
+      - image: codacy/git-version:1.0.8
```
Comment (andreaTP) [inline .circleci/config.yml]: I will use `latest`

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,120 +1,116 @@
...
+    working_directory: ~/workdir
+
+  circleci_job: &circleci_job
+    docker:
+      - image: circleci/circleci-cli:0.1.5705
+    working_directory: ~/workdir
+
+jobs:
+  checkout_and_version:
```
Comment (andreaTP) [inline .circleci/config.yml]: you can re-use this very same orb (an old version) for this task

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,120 +1,116 @@
...
+            /bin/git-version > .version
+      - run:
+          name: Current version
+          command: cat .version
+      - persist_to_workspace:
+          root: ~/
+          paths:
+            - workdir
+
+  tag_version:
```
Comment (andreaTP) [inline .circleci/config.yml]: as before, re-use this very same orb for this task

File: .circleci/config.yml [hunk 4]
```diff
@@ -1,120 +1,116 @@
...
+          command: git tag $(cat .version) && git push --tags
+
+  pack_and_validate:
+    <<: *circleci_job
+    steps:
+      - attach_workspace:
+          at: ~/
+      - run:
+          name: Pack orb
+          command: circleci config pack src > orb.yml
```
Comment (andreaTP) [inline .circleci/config.yml]: just to be extra-safe I would `rm -f orbl.yml` before

File: .circleci/config.yml (hunk 1 above)
Comment (h314to) [reply .circleci/config.yml]: Ok, that will save us the effort of having to bump it.

File: .circleci/config.yml [hunk 5]
```diff
@@ -1,120 +1,116 @@
...
-  orb-tools: circleci/orb-tools@8.5.0
+references:
+  git_job: &git_job
+    docker:
+      - image: codacy/git-version:1.0.8
+    working_directory: ~/workdir
+
+  circleci_job: &circleci_job
+    docker:
+      - image: circleci/circleci-cli:0.1.5705
```
Comment (h314to) [inline .circleci/config.yml]: Do you think i should use latest here too? I'm afraid they might break something without warning.

File: .circleci/config.yml (hunk 2 above)
Comment (h314to) [reply .circleci/config.yml]: ok, that simplifies it

File: .circleci/config.yml (hunk 4 above)
Comment (h314to) [reply .circleci/config.yml]: because i'm using a single > the content of the file (if it exists) is overwritten

Comment (andreaTP) [reply .circleci/config.yml]: you are right

File: .circleci/config.yml (hunk 5 above)
Comment (andreaTP) [reply .circleci/config.yml]: yep, keep latest only for things we own and control the release.

File: .circleci/config.yml (hunk 4 above)
Comment (andreaTP) [reply .circleci/config.yml]: correct, np!

**Comment by andreaTP** [comment]
@h314to can you check why there is a circle ci test missing?

File: .circleci/config.yml [hunk 6]
```diff
@@ -1,120 +1,79 @@
...
+          command: circleci orb publish orb.yml codacy/base@$(cat .version) --token $CIRCLE_TOKEN
 
 workflows:
-  # this `lint-pack_validate_publish-dev` workflow will run on any commit
-  lint-pack_validate_publish-dev:
+
+  pack_validate_publish:
     jobs:
-      # lint your destructured orb YAML files
-      - orb-tools/lint
```
Comment (andreaTP) [inline .circleci/config.yml]: argh I think this is the check that circle ci is waiting for, can you please re-enable it? cc @h314to

**Comment by andreaTP** [comment]
thanks @h314to !

**Comment by h314to** [comment]
thanks for the feedback :+1:

**Pull request merged**